### PR TITLE
Fixed and refactored GitHub link on pages from va.gov-team

### PR DESCRIPTION
**Pull request by U-DON**
## Description
- Refactored the link to "Edit this page on GitHub" so that its URL is determined at the data layer (GraphQL).
- Merged all of the [Gatsby Node API](https://www.gatsbyjs.com/docs/node-apis/) stuff from the `mdx-pages` plugin into `gatsby-node.js`.

## Testing done
Locally tested links on pages that represented three different sources:
- from the documentation site itself (`/getting-started`)
- from formation-react (`/visual-design/components/alertbox/`)
- from va.gov-team (`/platform/working-with-vsp/requesting-support`)

## Acceptance criteria
- [ ] GitHub links should work on pages pulled from va.gov team.

## Definition of done
- [ ] Changes have been tested in vets-website
- [ ] Changes have been tested in IE11, if applicable
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: packages/documentation/gatsby-node.js [hunk 1]
```diff
@@ -8,69 +6,226 @@
...
+
+exports.onCreatePage = helpers => {
+  setSourceUrl(helpers);
+};
+
+exports.onCreateWebpackConfig = ({ actions }) => {
+  actions.setWebpackConfig({
+    resolve: {
+      modules: [path.resolve(__dirname, 'src'), 'node_modules'],
+    },
```
Comment (U-DON) [inline packages/documentation/gatsby-node.js:229]: I pulled the stuff from the former `mdx-pages` plugin into here. Maybe it's cleaner to split into plugins, but not sure of best practice here. I just didn't think these functions did enough with the data to be considered "plugins".

Could be worth splitting out into plugins if we had more sources to pull from, but ultimately, I thought keeping our custom usage of all the Gatsby Node APIs in one place would be easier for discoverability.

File: packages/documentation/src/templates/page.js [hunk 2]
```diff
@@ -9,12 +9,8 @@ export default ({ location, pageContext }) => {
   const content = <div dangerouslySetInnerHTML={{ __html: html }} />;
 
   return (
-    <SidebarLayout location={location}>
-      <div>{content}</div>
-      <div>
-        <h5>Source:</h5>
-        <a href={sourceUrl}>{sourceUrl}</a>
-      </div>
+    <SidebarLayout location={location} pageContext={pageContext}>
```
Comment (U-DON) [inline packages/documentation/src/templates/page.js:12]: I acknowledge there's some duplication here. There might be a better way of combining some of these templates/layouts, but to keep this PR focused, I'll save that refactor for another day...a

File: packages/documentation/gatsby-config.js [hunk 3]
```diff
@@ -1,6 +1,3 @@
-/* eslint-disable */
-const path = require('path');
```
Comment (billfienberg) [inline packages/documentation/gatsby-config.js:2]: I'm assuming `path` wasn't used. If so, I'm surprised that wasn't caught by a linter.

After [searching the repo for `eslintrc`](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/search?q=eslintrc&unscoped_q=eslintrc), it looks like we actually [ignore `.eslintrc`](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/e0049086edb435e5a7881360504352269d7b05d0/.gitignore#L12), and unignore it only within the [`.gitignore` file for the `vagov-eslint` package](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/e0049086edb435e5a7881360504352269d7b05d0/packages/vagov-eslint/.gitignore).

Comment (U-DON) [reply packages/documentation/gatsby-config.js:2]: There are several places I've found that didn't seem to actually get linted or formatted by Prettier. That checks out...

File: packages/documentation/gatsby-node.js [hunk 4]
```diff
@@ -8,69 +6,226 @@
...
+  // Avoid setting source URLs that we can't confidently determine.
+  if (!match) return;
+
+  // Build the source URL once we've determined that the page is from src/pages.
+  const [relativePath] = match;
+  const sourceUrl = `${GITHUB_FILE_BASE_URL}${relativePath}`;
+
+  deletePage(page);
+
+  createPage({
```
Comment (billfienberg) [inline packages/documentation/gatsby-node.js:192]: Does creating the page after deleting it allow you to effectively add `context` to the page without creating a duplicate page?

File: packages/documentation/gatsby-node.js [hunk 5]
```diff
@@ -8,69 +6,226 @@
...
+    ...page,
+    context: {
+      ...page.context,
+      sourceUrl,
+    },
+  });
+};
+
+
+exports.onCreateNode = ({ node, getNode, actions }) => {
```
Comment (billfienberg) [inline packages/documentation/gatsby-node.js:202]: Moved from `packages/documentation/plugins/mdx-pages/gatsby-node.js` to here.

File: packages/documentation/gatsby-node.js [hunk 6]
```diff
@@ -8,69 +6,226 @@
...
+  if (node.internal.type === 'Mdx' && parent.sourceInstanceName) {
+    createNodeField({
+      node,
+      name: 'sourceInstanceName',
+      value: parent.sourceInstanceName,
+    });
+  }
+};
+
+exports.createPages = async helpers => {
```
Comment (billfienberg) [inline packages/documentation/gatsby-node.js:216]: Moved from `packages/documentation/plugins/mdx-pages/gatsby-node.js` to here.

File: packages/documentation/gatsby-node.js [hunk 7]
```diff
@@ -8,69 +6,226 @@
...
+  await createFormationReactPages(helpers);
+  await createVaGovTeamPages(helpers);
+};
+
+exports.onCreatePage = helpers => {
+  setSourceUrl(helpers);
+};
+
+exports.onCreateWebpackConfig = ({ actions }) => {
```
Comment (billfienberg) [inline packages/documentation/gatsby-node.js:225]: Moved from `packages/documentation/plugins/mdx-pages/gatsby-node.js` to here.

File: packages/documentation/src/templates/page.js (hunk 2 above)
Comment (billfienberg) [reply packages/documentation/src/templates/page.js:12]: I'll happily accept a small amount of duplication over a suboptimal abstraction. 

Reference: 
- https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction
- https://kentcdodds.com/blog/aha-programming#aha-

File: packages/documentation/gatsby-node.js (hunk 4 above)
Comment (U-DON) [reply packages/documentation/gatsby-node.js:192]: Yup, that appears to be the [official way of adding context](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#pass-context-to-pages).

File: packages/documentation/src/templates/page.js (hunk 2 above)
Comment (U-DON) [reply packages/documentation/src/templates/page.js:12]: For sure, and the refactor could just be renaming all of the templates in a morre consistent and clear way.

File: packages/documentation/src/layouts/SidebarLayout.jsx [hunk 8]
```diff
@@ -1,17 +1,17 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import Layout from './Layout';
 import Sidebar from '../components/sidebar/Sidebar';
-import PageFooter from '../components/pageFooter/PageFooter';
```
Comment (billfienberg) [inline packages/documentation/src/layouts/SidebarLayout.jsx:6]: It seems like the `PageFooter` component renders the `Components` and `Documentation` components. Are we losing any valuable navigation functionality by deleting the `PageFooter` component, and ergo the `Components` and `Documentation` components?

File: packages/documentation/gatsby-node.js (hunk 4 above)
Comment (billfienberg) [reply packages/documentation/gatsby-node.js:192]: Cool. Thanks for linking to the docs.

File: packages/documentation/src/layouts/SidebarLayout.jsx (hunk 8 above)
Comment (U-DON) [reply packages/documentation/src/layouts/SidebarLayout.jsx:6]: If you look into the `Components` and `Documentation` components, you'll see they implemented rather roundabout ways of rendering the edit link. And that was all `PageFooter` was really for.

Comment (billfienberg) [reply packages/documentation/src/layouts/SidebarLayout.jsx:6]: That makes sense. Thanks for the explanation.

**Pull request merged**